lab-kyler/greet.js

```diff
@@ -0,0 +1,9 @@
+'use strict';
+module.exports = {};
+
+if (process.argv[2]) console.log('Hello, ' + process.argv[2] + '!');
+
+module.exports.greet = function(name) {
+  if (arguments.length === 0) throw new Error('No name provided!');
+  return 'Hello, ' + name + '!';
+};
```

Comment:
Good use of process.argv[2] to take care of the bonus! Minor nitpick/suggestion, you should start using string interpolation to handle concatenation. For instance on line 4 you could: console.log(`Hello ${process.argv[2]}!`) and line 8 you could: return (`Hello ${name}`)